subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -566,6 +569,29 @@ static void hawkbit_dump_deployment(struct hawkbit_dep_res *d)
 	LOG_DBG("md5sum =%s", l->md5sum_http.href);
 }
 
+void hawkbit_set_server_addr(char * addr_str) {
+	if (!strcmp(addr_str, hawkbit_server)){
+		LOG_INF("hawkbit server address already corectly configured");
+	} else if(!is_hawkbit_init_done) {
```

Comment:
I'm not sure the way this check is currently implemented makes sense... There could be a race condition if the address is changed while hawkbit_init is running.
In general, there are no technical reasons to restrict it this far. The server could be changed anytime except from when start_http_client() is being run in line 1039 until cleanup_connection() in line 1230. This could also be simplified to "only allow changes when hawkbit_probe is not running".